builtin/providers/aws/data_source_availability_zones.go

```diff
@@ -0,0 +1,50 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceAwsAvailabilityZones() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsAvailabilityZonesRead,
+
+		Schema: map[string]*schema.Schema{
+			"available": &schema.Schema{
+				Type:     schema.TypeSet,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
```

Comment:
Certainly doesn't have to be in the first version of this data source, but I was thinking that when mapping `Describe*` calls onto data sources we'd map each of the filters onto an `Optional` attribute and then include them in the call.

In this case, that could be attributes `regions`, which would be a list of regions to constrain to, and `states`, allowing to filter by state.

So then I'd use it something like this:

```hcl
data "aws_availability_zones" "us" {
    regions = ["us-west-1", "us-west-2", "us-east-1"]
    states = ["available", "information", "impaired"] # don't include "unavailable" AZs
}
```